src/components/form/fields/birthdayField.jsx

```diff
@@ -135,18 +135,39 @@ const validateDay = day => {
     return 'birthdate.dayOfMonth';
 };
 
-const validateYear = year => {
-    if (isNumeric(year)) {
-        const yearNbr = parseInt(year);
-        const currentYear = new Date().getFullYear();
-        const age = currentYear - yearNbr;
-        if (age > 5 && age < 130) {
+const isLeapYear = year =>  ((year % 4 == 0) && (year % 100 != 0)) || (year % 400 ==0);
+
+const validateYear = year => isNumeric(year) ? false : 'birthdate.year';
+
+const validateCompatibilityMonthDay = (year, month, day) => {
```

Comment:
There is no type line Either un JS, unlike you implements your own in TS.
Eventaly you can return or throw an Error (exception) or maybe a Promise, but this is not an async context...
The cleanest would be a custom object with an attribute `type: "success"|"error"` and an additionnal property `description: string` if the type is error. Using TS and conditionnal type it would be good.